teammapper-frontend/src/app/modules/application/components/dialog-share/dialog-share.component.ts

```diff
@@ -72,20 +78,28 @@ export class DialogShareComponent implements OnInit {
 
     const newMap = await response.json();
     if (newMap && newMap.map.uuid) {
-      const sucessMessage = await this.utilsService.translate(
+      const successMessage = await this.utilsService.translate(
         'TOASTS.SUCCESSFULLY_DUPLICATED'
       );
-      this.toastrService.success(sucessMessage);
-
-      // Built in delay to allow users to read the toast (if we redirect immediately the toast gets swallowed up)
-      // The reason we're doing a client-side replace and not server-side redirect is to make sure all client-side data is refreshed
-      setTimeout(
-        () =>
-          window.location.replace(
-            `/map/${newMap.map.uuid}#${newMap.modificationSecret}`
-          ),
-        750
-      );
+
+      await this.storageService.set(newMap.map.uuid, {
+        adminId: newMap.adminId,
+        modificationSecret: newMap.modificationSecret,
+        ttl: newMap.map.deletedAt,
+        rootName: newMap.map.data[0].name,
+      });
+
+      const baseUrl = `/map/${newMap.map.uuid}`;
+      const fragment = newMap.modificationSecret;
+
+      const url = new URL(baseUrl, window.location.origin);
+
+      url.searchParams.set('showToast', 'true');
+      url.searchParams.set('toastMessage', successMessage);
+
+      url.hash = fragment;
+
+      window.location.replace(url.toString());
```

Comment:
same here, maybe use angular router instead of doing this manually?

Reply:
Added in now, the trick is to close the dialog manually before this.router.navigate()